chain/epoch_manager/src/lib.rs

```diff
@@ -288,59 +288,126 @@ impl EpochManager {
             }
             let block_stats = block_validator_tracker
                 .get(&(i as u64))
-                .unwrap_or_else(|| &ValidatorStats { expected: 0, produced: 0 });
-            // Note, validator_kickout_threshold is 0..100, so we use * 100 to keep this in integer space.
-            if block_stats.produced * 100
-                < u64::from(block_producer_kickout_threshold) * block_stats.expected
-            {
-                validator_kickout.insert(
-                    account_id.clone(),
-                    ValidatorKickoutReason::NotEnoughBlocks {
-                        produced: block_stats.produced,
-                        expected: block_stats.expected,
-                    },
-                );
-            }
+                .unwrap_or_else(|| &ValidatorStats { expected: 0, produced: 0 })
+                .clone();
             let mut chunk_stats = ValidatorStats { produced: 0, expected: 0 };
             for (_, tracker) in chunk_validator_tracker.iter() {
                 if let Some(stat) = tracker.get(&(i as u64)) {
                     chunk_stats.expected += stat.expected;
                     chunk_stats.produced += stat.produced;
                 }
             }
-            if chunk_stats.produced * 100
-                < u64::from(chunk_producer_kickout_threshold) * chunk_stats.expected
+            _total_stake += v.stake();
+            validator_block_chunk_stats
+                .insert(account_id.clone(), BlockChunkValidatorStats { block_stats, chunk_stats });
+        }
+
+        #[allow(unused_mut)]
+        let mut exempted_validators = HashSet::<&AccountId>::new();
+        #[cfg(feature = "protocol_feature_max_kickout_stake_ratio")]
+        if checked_feature!(
+            "protocol_feature_max_kickout_stake_ratio",
+            MaxKickoutStakeRatio,
+            epoch_info.protocol_version()
+        ) {
```

Comment:
Another way to feature-gate this is to make EpochConfig use `100` as `validator_max_kickout_stake_ratio` for old protocol versions, and whatever we want to in the latter protocol versions. In some sense, this is more correct, as `EpochConfig` more faithfully reflects realitiy. 

Do we care about `EpochConfig` making sense? (eg, do we serialize or hash it anywhere?)  If we do, I have a mild preference for gating via config, if we don't the current code feels simpler.